### DISCUSSION TITLE
How to use the state example

### DISCUSSION THREAD

**Discussion by HerrMuellerluedenscheid** [rwf2/Rocket#2182] [Questions] [answered]
When I use this example in the [state section](https://rocket.rs/v0.4/guide/state/):

```
#[macro_use] extern crate rocket_contrib;

use rocket_contrib::databases::diesel;

#[database("sqlite_logs")]
struct LogsDbConn(diesel::SqliteConnection);

fn main() {
    rocket::ignite()
       .attach(LogsDbConn::fairing())
       .launch();
}
```

it doesn't seem to find the `#[database("sqlite_logs")]` macro. What am I doing wrong?

**Comment by SergioBenitez** [comment, marked as answer]
You're probably reading outdated documentation. Are you using 0.4 or 0.5? Those docs are for 0.4. Use the version switcher to get to 0.5. Here's a direct link: https://rocket.rs/v0.5-rc/guide/state/#databases.

**Reply by HerrMuellerluedenscheid** [reply]
Thats it! Thanks for the immediate reply! And thanks for a wonderful framwork!
